src/library.js

```diff
@@ -657,7 +657,23 @@ addToLibrary({
 
     {{{ makeSetValue('daylight', '0', 'Number(winterOffset != summerOffset)', 'i32') }}};
 
-    var extractZone = (date) => date.toLocaleTimeString(undefined, {hour12:false, timeZoneName:'short'}).split(' ')[1];
+    var extractZone = (date) => {
+      var timezoneOffset = date.getTimezoneOffset();
+      if (isNaN(timezoneOffset) || timezoneOffset === 0) {
+        return `UTC`;
+      }
+
+      // Why inverse sign?
+      // Read here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTimezoneOffset
+      var sign = timezoneOffset >= 0 ? "-" : "+";
+
+      var absOffset = Math.abs(timezoneOffset)
+      var hours = String(Math.floor(absOffset / 60)).padStart(2, "0");
+      var minutes = String(absOffset % 60).padStart(2, "0");
+
+      return `UTC${sign}${hours}${minutes}`;
+    }
+
     var winterName = extractZone(winter);
     var summerName = extractZone(summer);
```

Comment:
```suggestion
    /**
     * Extracts the timezone from offset and returns it as UTC±hhmm
     *
     * If timezoneOffset equal to 0 returns "UTC"
     * @param {number} timezoneOffset
     * @return {string}
     */
    function extractZone(timezoneOffset) {
        if (timezoneOffset === 0) {
            return `UTC`;
        }
        // Why inverse sign?
        // Read here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTimezoneOffset
        var sign = timezoneOffset >= 0 ? "-" : "+";
    
        var absOffset = Math.abs(timezoneOffset)
        var hours = String(Math.floor(absOffset / 60)).padStart(2, "0");
        var minutes = String(absOffset % 60).padStart(2, "0");
    
        return `UTC${sign}${hours}${minutes}`;
    }
    var winterName = extractZone(winterOffset);
    var summerName = extractZone(summerOffset);
```

Reply:
Thanks @dvlezhnev updated the PR with those changes!